project_task_code/models/project_task.py

```diff
@@ -41,3 +41,17 @@ def name_get(self):
             name = "[{}] {}".format(rec.code, task[1])
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def name_search(self, name, args=None, operator="ilike", limit=100):
+        result = []
+
+        if name:
+            domain = [("code", operator, name)]
+            tasks = self.search(domain, limit=limit)
+            result = tasks.name_get()
+
+        return (
+            super().name_search(name, args=args, operator=operator, limit=limit)
+            + result
+        )
```

Comment:
Just expand the domain with the additional search condition.
Suggested pseudocode:
```suggestion
    def _name_search(self, name, args=None, operator="ilike", limit=100, name_get_uid=None):
        new_args = expression.OR([args or [], [("code", operator, name)]])
        return super()._name_search(name, new_args, operator, limit, name_get_uid)
```